persisted_ai/rag_activities_integration_test.go

```diff
@@ -41,7 +41,27 @@ func setupTestWorkspace(t *testing.T, ctx context.Context) (string, string) {
 		}
 	}
 
-	require.NotEmpty(t, workspaceId, "No workspace found for repository root: %s", cleanedRepoRoot)
+	if workspaceId == "" {
+		// If no workspace found, try finding the git source directory
+		cmd := exec.Command("git", "rev-parse", "--git-common-dir")
+		gitCommonDirBytes, err := cmd.Output()
+		if err == nil {
+			gitCommonDir := strings.TrimSpace(string(gitCommonDirBytes))
+			if !filepath.IsAbs(gitCommonDir) {
+				gitCommonDir = filepath.Join(cleanedRepoRoot, gitCommonDir)
+			}
+			gitSourceDir := filepath.Dir(gitCommonDir)
+			// Check if there's a workspace for the source directory
+			for _, ws := range workspaces {
+				if filepath.Clean(ws.LocalRepoDir) == filepath.Clean(gitSourceDir) {
+					workspaceId = ws.Id
+					break
+				}
+			}
+		}
+	}
+
+	require.NotEmpty(t, workspaceId, "No workspace found for repository root or git source directory")
```

Comment:
Essentially: initializing a workspace for the repo where you do dev work for sidekick is required to run these integration tests, since we rely on that to reuse cached embeddings without having to embed again and again every time. Maybe we can just initialize it automatically though on first run, so we don't require the human to do it.

Reply:
Ok I did that and realized it was actually a bad idea, explained why in comments and instead improved the error message to prompt the dev.